src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +19,55 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY - ampHours_ ;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+
+
+    if(previousCurrent_ < 0)
+    {
     return false;
+    }
+    else
+    {
+       return true;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+
+    double timeRemaining = (BATTERY_AMP_HOUR_CAPACITY-ampHours_) / previousCurrent_ ;
+
+    timeRemaining = qAbs(timeRemaining) / MILISECONDS_TO_HOURS;
+
+    return QTime(0, 0, 0, 0).addMSecs(timeRemaining);
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
-    // Update your variables here.
+
+
+    double presentCurrent = batteryData.current;
+    if(previousCurrent_ == 0)
+    {
+        aveCurrent_ = presentCurrent;
+    }
+    else
+    {
+        aveCurrent_ = (previousCurrent_+presentCurrent)/2;
```

Comment:
Operators should have spaces around them, so:
`(previousCurrent_ + presentCurrent) / 2;`